test.js

```diff
@@ -1,85 +1,243 @@
 import test from 'node:test'
 import assert from 'node:assert'
 import timers from 'node:timers/promises'
-import { CancelStuckTransactions } from './index.js'
+import {
+  StuckTransactionsCanceller,
+  cancelTx,
+  getRecentSendMessage
+} from './index.js'
 
-test('CancelStuckTransactions', async () => {
-  const tx = {
-    hash: 'hash',
-    maxPriorityFeePerGas: 10n,
-    nonce: 20,
-    from: '0x0'
-  }
-  const storage = new Map()
-  const sentTransactions = []
-  const cancelStuckTransactions = new CancelStuckTransactions({
-    store ({ hash, timestamp, from, maxPriorityFeePerGas, nonce }) {
-      assert(!storage.has(hash))
-      assert.strictEqual(typeof hash, 'string')
-      assert.strictEqual(typeof timestamp, 'string')
-      assert.strictEqual(typeof from, 'string')
-      assert.strictEqual(typeof maxPriorityFeePerGas, 'bigint')
-      assert.strictEqual(typeof nonce, 'number')
-      storage.set(hash, {
-        hash,
-        timestamp,
-        from,
-        maxPriorityFeePerGas,
-        nonce
+test('StuckTransactionsCanceller', async t => {
+  await t.test('#addPending()', async () => {
+    const tx = {
+      hash: 'hash',
+      maxPriorityFeePerGas: 10n,
+      nonce: 20,
+      from: '0x0'
+    }
+    const storage = new Map()
+    const stuckTransactionsCanceller = new StuckTransactionsCanceller({
+      store: {
+        add ({ hash, timestamp, from, maxPriorityFeePerGas, nonce }) {
+          assert(!storage.has(hash))
+          assert.strictEqual(typeof hash, 'string')
+          assert.strictEqual(typeof timestamp, 'string')
+          assert.strictEqual(typeof from, 'string')
+          assert.strictEqual(typeof maxPriorityFeePerGas, 'bigint')
+          assert.strictEqual(typeof nonce, 'number')
+          storage.set(hash, {
+            hash,
+            timestamp,
+            from,
+            maxPriorityFeePerGas,
+            nonce
+          })
+        },
+        list () {
+          throw new Error('Should not be called')
+        },
+        remove (hash) {
+          throw new Error('Should not be called')
+        }
+      },
+      log: () => {
+        // TODO: Test logs
```

Comment:
I have good experience with using `debug` for test logs. Then, I can run `DEBUG=test node --test` to get verbose logs from the tests.

Reply:
Great idea!